pull-requests/code-reviews/recipes/TypeScript.md

```diff
@@ -1,92 +1,111 @@
 # TypeScript Code Reviews
 
-## Style Guide
+TypeScript projects should have linting configured, and should refuse PR's that do not comply with
+the project's linting standard.
 
-We use [`Prettier`](https://prettier.io/) to do code formatting for TypeScript. Using an automated code formatting tool like Prettier enforces a well accepted style guide that was collaboratively built by a wide range of companies including Microsoft, Facebook, and AirBnB. It is highly recommended that you use one of the native editor integrations and enable format on save. In addition, by default, we use the following overrides (shown here in the config format for VSCode) to standardize on single quotes and a 4-space indent:
+## Setting up [ESLint](https://eslint.org/docs/about/)
 
-```json
-{
-    ...
-    "prettier.singleQuote": true,
-    "prettier.tabWidth": 4
-}
+Per guidance outlined in [Palantir's 2019 TSLint roadmap](https://medium.com/palantir/tslint-in-2019-1a144c2317a9),
+TypeScript code should be linted with [ESLint](https://github.com/eslint/eslint). Resources for
+linting TypeScript code with ESLint can be found in the [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)
+repository.
+
+To [install and configure linting with ESLint](https://github.com/typescript-eslint/typescript-eslint/tree/master/docs/getting-started/linting),
+install the following dependencies:
+
+```bash
+npm i -D eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin
 ```
 
-## Linter
+Add a `.eslintrc.js` to the root of your project:
+
+```javascript
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  plugins: [
+    '@typescript-eslint',
+  ],
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+  ],
+};
+```
 
-All developers should run `TSLint` in a pre-commit hook to ensure standard formatting. We highly recommend using an editor integration like [vscode-tslint](https://github.com/Microsoft/vscode-tslint) to provide realtime feedback.
+Add a `.eslintignore` with files & folders to omit:
 
-Recommended: [lint-staged and husky](https://github.com/okonet/lint-staged#installation-and-setup)
+```ignore
+# don't ever lint node_modules
+node_modules
+# don't lint build output
+dist
+# don't lint nyc coverage
+coverage
+```
```

Comment:
A better (default) solution might be to point the `--ignore-path` to the project's `.gitignore` file as explained here: https://eslint.org/docs/2.13.1/user-guide/configuring#using-an-alternate-file